src/tests/Common/helixpublishwitharcade.proj

```diff
@@ -328,21 +345,34 @@
       <PayloadGroup>$([MSBuild]::MakeRelative($(PayloadsRootDirectory), %(FullPath)))</PayloadGroup>
       <PayloadDirectory>%(FullPath)</PayloadDirectory>
       <XUnitWrapperDlls>$([System.String]::Join(' ', $([System.IO.Directory]::GetFiles(%(FullPath), '*.XUnitWrapper.dll', SearchOption.AllDirectories))).Replace($([MSBuild]::EnsureTrailingSlash(%(FullPath))),''))</XUnitWrapperDlls>
+      <RuntimeTestApks>$([System.String]::Join(' ', $([System.IO.Directory]::GetFiles(%(FullPath), '*.apk', SearchOption.AllDirectories))))</RuntimeTestApks>
     </Payloads>
 
     <Payloads Update="@(Payloads)">
       <TestGroup>%(PayloadGroup)</TestGroup>
       <!-- When Payload contains more than one *.XUnitWrapper.dll TestGroup should not be specified. -->
       <TestGroup Condition=" $([MSBuild]::ValueOrDefault(%(XUnitWrapperDlls), '').IndexOf('.XUnitWrapper.dll')) != $([MSBuild]::ValueOrDefault(%(XUnitWrapperDlls), '').LastIndexOf('.XUnitWrapper.dll')) "></TestGroup>
+      <PayloadZipFile>$(PayloadsRootDirectory)\%(PayloadGroup).zip</PayloadZipFile>
     </Payloads>
 
-    <HelixWorkItem Include="@(Payloads->Metadata('PayloadGroup'))">
+    <HelixWorkItem Include="@(Payloads->Metadata('PayloadGroup'))" Condition=" '$(IsRunningOnMobileTargets)' == 'false' ">
       <PayloadDirectory>%(PayloadDirectory)</PayloadDirectory>
       <Command>dotnet $(XUnitRunnerDll) %(XUnitWrapperDlls) $(XUnitRunnerArgs)</Command>
       <Command Condition=" '%(TestGroup)' != '' ">dotnet $(XUnitRunnerDll) %(XUnitWrapperDlls) $(XUnitRunnerArgs) -trait TestGroup=%(TestGroup)</Command>
       <Timeout Condition=" '$(TimeoutPerTestCollectionInMinutes)' != '' ">$([System.TimeSpan]::FromMinutes($(TimeoutPerTestCollectionInMinutes)))</Timeout>
     </HelixWorkItem>
 
+    <XHarnessApkToTest Include="@(Payloads->Metadata('PayloadZipFile'))" Condition="'$(TargetOS)' == 'Android'">
+      <TestTarget Condition="'$(TargetArchitecture)' == 'arm'">armeabi-v7a</TestTarget>
+      <TestTarget Condition="'$(TargetArchitecture)' == 'arm64'">arm64-v8a</TestTarget>
+      <TestTarget Condition="'$(TargetArchitecture)' == 'x64'">x86_64</TestTarget>
+      <TestTarget Condition="'$(TargetArchitecture)' == 'x86'">x86</TestTarget>
+      <TestTimeout Condition=" '$(TimeoutPerTestCollectionInMinutes)' != '' ">$([System.TimeSpan]::FromMinutes($(TimeoutPerTestCollectionInMinutes)))</TestTimeout>
+      <LaunchTimeout>00:05:00</LaunchTimeout>
```

Comment:
`LaunchTimeout` also doesn't do anything since in the `CustomCommands`, you're not utilizing the `$launch_timeout` variable. You would need to add the `--launch-timeout` option to your XHarness commands.

Reply:
Same goes for `TestTimeout`

Reply:
removed.